src/main/java/org/springframework/data/couchbase/core/convert/MappingCouchbaseConverter.java

```diff
@@ -145,8 +142,14 @@ public MappingCouchbaseConverter(
 			final MappingContext<? extends CouchbasePersistentEntity<?>, CouchbasePersistentProperty> mappingContext,
 			final String typeKey) {
 		super(new DefaultConversionService());
-
 		this.mappingContext = mappingContext;
+		// this is how the MappingCouchbaseConverter gets the custom conversions.
+		// the conversions Service gets them in afterPropertiesSet()
+		CustomConversions customConversions = new CouchbaseCustomConversions(Collections.emptyList());
+		this.setCustomConversions(customConversions);
+		// if the mappingContext does not have the SimpleTypes, it will not know that they have converters, then it will
+		// try to access the fields of the type and (maybe) fail with InaccessibleObjectException
+		((CouchbaseMappingContext) mappingContext).setSimpleTypeHolder(customConversions.getSimpleTypeHolder());
```

Comment:
While this would work, associating custom conversions should happen outside the converter. Typically, `CustomConversions` gets configured by users, and then it should get associated with the converter/mapping context within `@Bean` methods.

Reply:
They were already populated in AbstractCouchbaseConverter.
```
	protected CustomConversions conversions = new CouchbaseCustomConversions(Collections.emptyList());
```
I set them again in the MappingCouchbaseConverter constructor so that the mappingContext could capture the SimpleTypes so those converters would be used. 

btw - these aren't strictly user-supplied "custom" conversions. These are provided/required by couchbase (and also  included in any customer-created CouchbaseCustomConversions).